cli/genkey/genkey_test.go

```diff
@@ -1 +1,45 @@
 package genkey
+
+import (
+	"encoding/json"
+	"github.com/cloudflare/cfssl/cli"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+func TestGenkey(t *testing.T) {
+	//testing through console
+	gopath := os.Getenv("GOPATH")
+	cfssl := path.Join(gopath, "bin", "cfssl")
+	testdata := path.Join(gopath, "src", "github.com", "cloudflare", "cfssl", "testdata")
+
+	out1, err := exec.Command(cfssl, "genkey", path.Join(testdata, "csr.json")).Output()
+	var response map[string]interface{}
+	json.Unmarshal(out1, &response)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	if response["key"] == nil {
+		t.Fatal("No key is outputted.")
+	}
+	if response["csr"] == nil {
+		t.Fatal("No csr is outputted.")
+	}
+
+	c := cli.Config{}
+
+	err1 := genkeyMain([]string{path.Join(testdata, "csr.json")}, c)
```

Comment:
I guess you have tried to write "err :=" here, which would be caught by compiler as error. ":=" means create a new variable, "=" is a simple assignment. Since err is already created, use "=" here. Again, don't use "err1" or "err2". variable with number suffix is always bad, in practice.